src/Foundation/NSItemProvider.cs

```diff
@@ -0,0 +1,29 @@
+using System;
+using System.Threading.Tasks;
+using XamCore.CloudKit;
+
+namespace XamCore.Foundation 
+{
+#if MONOMAC && XAMCORE_2_0 // Only 64-bit on mac
+	public partial class NSItemProvider
+	{
+#if !XAMCORE_4_0
+		public unsafe virtual void RegisterCloudKitShare (Action<CloudKitRegistrationPreparationHandler> preparationHandler)
+		{
+			CloudKitRegistrationPreparationAction action = handler => preparationHandler (handler);
+			RegisterCloudKitShare (action);
+		}
+#endif
+		
+		public unsafe virtual Task<CloudKitRegistrationPreparationHandler> RegisterCloudKitShareAsync ()
```

Comment:
I just copied the generated method signature.

Is add/removing unsafe not an API breaking change @rolfbjarne?